pkg/agent/openflow/client.go

```diff
@@ -279,6 +280,18 @@ type Client interface {
 		dstIP net.IP,
 		outPort uint32,
 		igmp ofutil.Message) error
+
+	// InstallTrafficControlMarkFlows installs the flows to mark the packets for a traffic control rule.
+	InstallTrafficControlMarkFlows(name string, sourcePorts []uint32, targetPort uint32, direction v1alpha2.Direction, action v1alpha2.Action) error
+
+	// UninstallTrafficControlMarkFlows removes the flows for a traffic control rule.
+	UninstallTrafficControlMarkFlows(name string) error
+
+	// InstallTrafficControlCommonFlows installs the common flows for the target and return ports of traffic control rules.
+	InstallTrafficControlCommonFlows(targetPort, returnPort uint32) error
```

Comment:
Not sure if I understand why targetPort needs to be in this method.

Interfaces in my mind are:
```
InstallTrafficControlFlows(name string, sourcePorts []uint32, targetPort uint32, direction v1alpha2.Direction, action v1alpha2.Action)

UninstallTrafficControlFlows(name string)

AddTrafficControlReturnPort(port uint32)

DeleteTrafficControlReturnPort(port uint32)
```

Reply:
Updated as your description.